neqo-transport/src/connection/mod.rs

```diff
@@ -1531,119 +1569,121 @@ impl Connection {
         self.capture_error(Some(path), now, 0, res).ok();
     }
 
-    fn input_path(&mut self, path: &PathRef, d: &Datagram, now: Instant) -> Res<()> {
-        let mut slc = &d[..];
-        let mut dcid = None;
+    fn input_path(&mut self, path: &PathRef, d: Datagram<&[u8]>, now: Instant) -> Res<()> {
+        for mut slc in d.iter_segments() {
```

Comment:
I have to say, this is too much for me.  The idea that a datagram is actually multiple datagrams is not something I'm comfortable with.

Making this function more complicated is less than ideal as well.